api/v1beta1/kyma_types.go

```diff
@@ -375,28 +366,29 @@ func init() {
 	SchemeBuilder.Register(&Kyma{}, &KymaList{})
 }
 
-func (kyma *Kyma) UpdateCondition(reason KymaConditionReason, status metav1.ConditionStatus) {
+func (kyma *Kyma) UpdateCondition(conditionType KymaConditionType, status metav1.ConditionStatus) {
```

Comment:
be aware: breaking change for existing kymas, so backwards check has to be done with v1beta1 from before and after! this also needs to be back ported to `v1alpha1`

Reply:
@jakobmoellersap I agree with the breaking change. Means we should remove all conditions which have Type `Ready`, since with the old KLM a module could have been in error state, then new KLM will be deployed, module gets ready but will have a new own conditions. Thus, an orphaned condition `Ready` = false will be left. Such condition should be removed.

But why do we need a backport? Since this does not change the api of Kyma v1alpha1, but also fixes a bug of it. The Kyma CRD did not change at all

Reply:
understood. yes that makes sense we should probably introduce a compat package for this and then note that it should be removed after the next apiversion

Reply:
I will set the `Conditions` to nil, when converting from or to `v1alpha1`, since the Conditions will not be sued for STate Handling, but they will be updated in each reconcile loop. Thus, when converting from `v1alpha1` the one `Ready` condition will be removed, and replaced by the three current in the first reconcile loop

Reply:
But this only takes care of APIVersion upgrades. What if there is an existing v1beta1 CR that already has the wrong state?